pkg/trace/metrics/timing/timing.go

```diff
@@ -0,0 +1,143 @@
+// Package timing is used to aggregate timing calls within hotpaths to avoid using
+// repeated statsd calls. The package has a default set that reports at 10 second
+// intervals and can be used directly. If a different behaviour or reporting pattern
+// is desired, a custom Set may be created.
+package timing
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/atomic"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+)
+
+// AutoreportInterval specifies the interval at which the default set reports.
+const AutoreportInterval = 10 * time.Second
+
+var (
+	defaultSet = NewSet()
+	stopReport = defaultSet.Autoreport(AutoreportInterval)
+)
+
+// Since records the duration for the given metric name as time passed since start.
+// It uses the default set which is reported at 10 second intervals.
+func Since(name string, start time.Time) { defaultSet.Since(name, start) }
+
+// Flush reports the default set. It can be useful to call when the program
+// exits prematurely.
+func Flush() { stopReport() }
```

Comment:
Should this be called `Stop`? Flush sounds like something you can call multiple times, but it looks like you can really only call it once.

Reply:
I agree. Changed it.